internal/coreinternal/golden/testdata/sort-datapoint-slice/before.yaml

```diff
@@ -0,0 +1,52 @@
+resourceMetrics:
+  - resource:
+    scopeMetrics:
+      - metrics:
+          - description: The number of bytes which have flowed through the network interface.
+            name: nsxt.node.network.io
+            sum:
+              aggregationTemporality: 2
+              dataPoints:
+                - asInt: "0"
+                  attributes:
+                    - key: a
+                      value:
+                        stringValue: AAAA
+                  startTimeUnixNano: "1111111111111111111"
+                  timeUnixNano: "1111111111111111111"
+                - asInt: "0"
+                  attributes:
+                    - key: z
+                      value:
+                        stringValue: DDDD
+                    - key: x
+                      value:
+                        stringValue: InnerAttributeSort
+                  startTimeUnixNano: "1111111111111111111"
+                  timeUnixNano: "1111111111111111111"
+                - asInt: "0"
+                  attributes:
+                    - key: c
+                      value:
+                        stringValue: CCCC
+                  startTimeUnixNano: "1111111111111111111"
+                  timeUnixNano: "1111111111111111111"
+                - asInt: "0"
+                  attributes:
+                    - key: b
+                      value:
+                        stringValue: BBBBFirst
+                  startTimeUnixNano: "1111111111111111111"
+                  timeUnixNano: "1111111111111111111"
+                - asInt: "0"
+                  attributes:
+                    - key: b
+                      value:
+                        stringValue: BBBBSecond
```

Comment:
This PR doesn't incorporate attribute values, right?

Reply:
It does not, I just left that there so the tester can see how it interacts when the sorting of the slices encounters two of the same keys. In essence showing the first one it encounters gets priority in the sort.